sounds.js

```diff
@@ -41,41 +44,181 @@ const soundFiles = {
   audioBufferSourceNode.buffer = soundFileAudioBuffer;
-  audioBufferSourceNode.connect(audioContext.gain);
+  if (option === undefined) {
+    // play the sound globally
+    audioBufferSourceNode.connect(masterOut);
+  } 
+  else {
+    // play positional audio
+    const pannerNode = audioContext.createPanner();
+    pannerNode.panningModel = "HRTF";
+    const gainNode = audioContext.createGain();
+
+    audioBufferSourceNode.connect(pannerNode);
+    pannerNode.connect(gainNode);
+    gainNode.connect(masterOut);
+
+    const refDistance = option.refDistance !== undefined ? option.refDistance : 10;
+    const maxDistance = option.maxDistance !== undefined ? option.maxDistance : 50;
+    const distanceModel = option.distanceModel !== undefined ? option.distanceModel : 'inverse';
+    const volume = option.volume !== undefined ? option.volume : 1;
+
+    pannerNode.refDistance = refDistance;
+    pannerNode.maxDistance = maxDistance;
+    pannerNode.distanceModel = distanceModel;
+    gainNode.gain.value = volume;
+    gainNode.inReverbZone = false;
+
+    // handel currentPlayingSound array
+    audioBufferSourceNode.info = {voicer: option.voicer, panner: pannerNode, gainNode: gainNode};
+    currentPlayingSound.push(audioBufferSourceNode.info);
+    // remove audioBufferSourceNode from array when sound end
+    audioBufferSourceNode.addEventListener('ended', () => {
+      const index = currentPlayingSound.indexOf(audioBufferSourceNode.info);
+      if (index > -1) {
+        currentPlayingSound.splice(index, 1);
+      }
+    });
+  }
   audioBufferSourceNode.start(0, offset, duration);
   return audioBufferSourceNode;
 };
-const playSoundName = name => {
+const playSoundName = (name, option) => {
   const snds = soundFiles[name];
   if (snds) {
     const sound = snds[Math.floor(Math.random() * snds.length)];
-    playSound(sound);
+    playSound(sound, option);
     return true;
   } else {
     return false;
   }
 };
+
+
+function easeOfNoise (x, powNum) {
+  return Math.pow(x, powNum);
+}
+function createNoiseBuffer(audioContext, length) {
+
+  const bufferSize = length * audioContext.sampleRate;
+  const lBuffer = new Float32Array(bufferSize);
+  const rBuffer = new Float32Array(bufferSize);
+  
+  for (let i = 0; i < bufferSize; i++) {
+    const ratio = (bufferSize - i) / bufferSize; // will be 1 at the start of the loop and 0 at the end
+    const fadeAmount = easeOfNoise(ratio, 2);
+    lBuffer[i] = (1 - (2 * Math.random())) * fadeAmount;
+    rBuffer[i] = (1 - (2 * Math.random())) * fadeAmount;
+  }
+  
+  const buffer = audioContext.createBuffer(2, bufferSize, audioContext.sampleRate);
+  buffer.copyToChannel(lBuffer,0);
+  buffer.copyToChannel(rBuffer,1);
+
+  return buffer;
+
+}
+
+const upVectore = new THREE.Vector3(0, 1, 0);
+let reverbZonsPos = new THREE.Vector3(); 
+let cameraDirection = new THREE.Vector3();
+let localVector = new THREE.Vector3();
+const update = (timestamp, timeDiff) =>{
+  let inReverbZone = false;

```

Comment:
`isCameraInReverbZone()` should be a function.

Reply:
[isCameraInReverbZone function](https://github.com/webaverse/app/pull/3056/commits/2e5a63e97efd2cebc8a41f0040c666095dd0c533)